Description: do not crash when attempting test with no suitable dictionaries
 getTestDictionaryObject may return NULL
Author: Damyan Ivanov <dmn@debian.org>
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=480003

--- a/src/exam.cpp
+++ b/src/exam.cpp
@@ -82,6 +82,8 @@ GbgofficeExam::GbgofficeExam(TranslatorM
 
 	reset();
 	dict = trm->getTestDictionaryObject(0, 0);
+	if (dict == NULL) return;
+
 	dict->separateMeanings = true;
 	dict->boldDecoration = false;
 	dict->htmlOutput = false;
@@ -168,6 +170,8 @@ void GbgofficeExam::on_new_button()
 		}
 		dict = trm->getTestDictionaryObject(combo1.get_active_row_number(), 
 				combo2.get_active_row_number());
+		if (dict == NULL) return;
+
 		dict->separateMeanings = true;
 		dict->boldDecoration = false;
 		dict->htmlOutput = false;
